File: app/src/main/java/com/recursia/popularmovies/data/db/mappers/EntityToCastDatabaseModelMapper.java
package com.recursia.popularmovies.data.db.mappers;

import com.recursia.popularmovies.data.models.CastDatabaseModel;
import com.recursia.popularmovies.domain.models.Cast;
import com.recursia.popularmovies.utils.Mapper;

public class EntityToCastDatabaseModelMapper extends Mapper<Cast, CastDatabaseModel> {
    @Override
    public CastDatabaseModel transform(Cast cast) {
        CastDatabaseModel castDatabaseModel = new CastDatabaseModel();
        castDatabaseModel.setCastId(cast.getCastId());
        castDatabaseModel.setCharacter(cast.getCharacter());
        castDatabaseModel.setCreditId(cast.getCreditId());
        castDatabaseModel.setGender(cast.getGender());
        castDatabaseModel.setId(cast.getId());
        castDatabaseModel.setName(cast.getName());
        castDatabaseModel.setProfilePath(cast.getProfilePath());
        castDatabaseModel.setMovieId(cast.getMovieId());
        castDatabaseModel.setOrder(cast.getOrder());
        return castDatabaseModel;
    }
}

File: app/src/main/java/com/recursia/popularmovies/data/mappers/EntityToCastDatabaseModelMapper.kt
package com.recursia.popularmovies.data.mappers

import com.recursia.popularmovies.data.models.CastDatabaseModel
import com.recursia.popularmovies.domain.models.Cast
import com.recursia.popularmovies.utils.Mapper

class EntityToCastDatabaseModelMapper : Mapper<Cast, CastDatabaseModel>() {
    override fun transform(cast: Cast): CastDatabaseModel {
        val castDatabaseModel = CastDatabaseModel()
        castDatabaseModel.castId = cast.castId
        castDatabaseModel.character = cast.character
        castDatabaseModel.creditId = cast.creditId
        castDatabaseModel.gender = cast.gender
        castDatabaseModel.id = cast.id
        castDatabaseModel.name = cast.name
        castDatabaseModel.profilePath = cast.profilePath
        castDatabaseModel.movieId = cast.movieId
        castDatabaseModel.order = cast.order
        return castDatabaseModel
    }
}
